rust/flatbuffers/src/endian_scalar.rs

```diff
@@ -17,6 +17,24 @@
 
 use core::mem::size_of;
 
+mod private {
+    /// Types that are trivially transmutable are those where any combination of bits
+    /// represents a valid value of that type
+    ///
+    /// For example integral types are TriviallyTransmutable as all bit patterns are valid,
+    /// however, `bool` is not trivially transmutable as only `0` and `1` are valid
+    pub trait TriviallyTransmutable {}
```

Comment:
This acts as a sanity check on methods such as read_scalar, and friends which use copy_non_overlapping to perform a form of memory transmutation

Reply:
Nice, good use of the sealed trait pattern